src/shared/components/FormComponents/SelectController/SelectController.tsx

```diff
@@ -71,55 +71,68 @@ export const SelectController = <T extends FieldValues>({
     value,
     itemDisabled,
     icon,
-    withoutKey,
     hidden,
+    tooltip,
+    tooltipPlacement,
   }: GetMenuItem) => (
     <StyledMenuItem
-      {...(!withoutKey && { key: labelKey })}
+      key={labelKey}
       uiType={uiType}
       value={value as string}
       disabled={itemDisabled}
       className={hidden ? 'hidden-menu-item' : ''}
+      component={forwardRef((props, ref) => {
```

Comment:
I know that this is a suggestion I made to you, and I'm critiquing my own code here, but using the `forwardRef` and an inline function here, inside the `SelectController` body, means that we're recreating the `StyledMenuItem` (or rather, the component the `StyledMenuItem` will render as) on every render.

I wonder if we can't move this to a separately defined component, maybe like so:

```tsx
const SelectControllerMenuItem = forwardRef<
  HTMLLIElement,
  {
    disabled?: boolean;
    tooltip?: React.ReactNode;
    tooltipPlacement?: TooltipProps['placement'];
  } & React.ComponentPropsWithoutRef<'li'>
>(({ disabled, tooltip, tooltipPlacement, ...props }, ref) => {
  const item = <StyledItem itemDisabled={disabled} ref={ref} selectDisabled={disabled} {...props} />;

  return tooltip ? (
    <Tooltip
      tooltipTitle={tooltip}
      placement={tooltipPlacement}
      children={disabled ? <li {...props} /> : item}
      enterNextDelay={200}
    />
  ) : item;
});
```

... and then update the `component` prop to pass a stable reference, like so:

```suggestion
      component={SelectControllerMenuItem}
      tooltip={tooltip}
      tooltipPlacement={tooltipPlacement}
```

Before you try this change, I should warn you: I tried this locally, but I ran into some issues with prop type validation when using the `component` prop in combination with a `StyledComponent` wrapped around a Material-UI component — This isn't the [first time either](https://github.com/ChildMindInstitute/mindlogger-admin/pull/1714#discussion_r1599152417). 

Just as a heads up:

I think it's a worthwhile change, so I wanted it noted _somewhere_, buutt...

I'm also not going to block approval of this PR based on it, in case you decide not to try and address it (I suspect it's really a symptom of a larger configuration problem between Emotion and Material UI).

Reply:
This approach works and is somewhat more readable, thanks for the suggestion. I was able to get around the prop type validation but now we're relying on `shouldForwardProp` which can hide behavior a bit.